crates/volta-core/src/tool/package/resolve.rs

```diff
@@ -110,17 +109,15 @@ pub struct PackageIndex {
 ///
 /// * normal package installation from the public npm repo
 /// * installing packages from alternate registries, configured via .npmrc
-fn npm_view_query(name: &str, version: &str) -> Fallible<PackageIndex> {
-    let mut command = npm_view_command_for(name, version);
+fn npm_view_query(name: &str, version: &str, session: &mut Session) -> Fallible<PackageIndex> {
```

Comment:
What requires `Session` to be `&mut` instead of `&` here?

Reply:
The `PlatformSpec::checkout` operation requires a mutable reference because it can download new versions of Node and Yarn, which requires updating the Collections to include the new versions. I believe everywhere that `Session` is passed around, it’s a mutable reference, because in order to support the mutability anywhere, we need to support it everywhere.

Reply:
Ah, that makes sense. Thanks for clarifying!